integration/map_integration_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/storage/testdb"
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/testonly/integration"
+
+	_ "github.com/google/trillian/crypto/keys/der/proto" // Register PrivateKey ProtoHandler
+	stestonly "github.com/google/trillian/storage/testonly"
+)
+
+var singleTX = flag.Bool("single_transaction", false, "Experimental: whether to update the map in a single transaction")
+
+func TestInProcessMapIntegration(t *testing.T) {
```

Comment:
`integration/maptest/map_test.go` also defines an in-process map integration test harness. 
Do you think this test could be refactored in that style?  It would enable this test to be reused for remote map implementations as well.

Reply:
Possibly in a different PR, but I don't think this one quite matches up with the table-driven tests in there.